keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ApplicationController.kt

```diff
@@ -69,6 +78,36 @@ class ApplicationController(
     }
   }
 
+  @PostMapping(
+    path = ["/{application}/environment/{environment}/constraint"],
+    consumes = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
+    produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  @PreAuthorize("@authorizationSupport.userCanModifyApplication(#application)")
```

Comment:
Cool! I'm getting started on #816 right now -- this gives me a good example to follow! 👍

Reply:
P.S. It'd be nice to add tests for the pass/fail auth check.

Reply:
Done but only for failing, because that's the only relevant thing